native/dgetri.go

```diff
@@ -0,0 +1,86 @@
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dgetri computes the inverse of the matrix A using the LU factorization computed
+// by Dgetrf. On entry, a contains the PLU decomposition of A as computed by
+// Dgetrf and on exit contains the reciprocal of the original matrix.
+//
+// Work is temporary storage, and lwork specifies the usable memory length.
+// At minimum, lwork >= n and this function will panic otherwise.
+// Dgetri is a blocked inversion, but the block size is limited
+// by the temporary space available. If lwork == -1, instead of performing Dgetri,
+// the optimal work length will be stored into work[0].
+func (impl Implementation) Dgetri(n int, a []float64, lda int, ipiv []int, work []float64, lwork int) bool {
+	checkMatrix(n, n, a, lda)
+	if len(ipiv) < n {
+		panic(badIpiv)
+	}
+	nb := impl.Ilaenv(1, "DGETRI", " ", n, -1, -1, -1)
+	if lwork == -1 {
+		work[0] = float64(n * nb)
+		return true
+	}
+	if lwork < n {
+		panic(badWork)
+	}
+	if len(work) < lwork {
+		panic(badWork)
+	}
+	if n == 0 {
+		return true
+	}
+	ok := impl.Dtrtri(blas.Upper, blas.NonUnit, n, a, lda)
+	if !ok {
+		return false
+	}
+	nbmin := 2
+	ldwork := nb
+	if nb > 1 && nb < n {
+		iws := max(ldwork*n, 1)
+		if lwork < iws {
+			nb = lwork / ldwork
+			nbmin = max(2, impl.Ilaenv(2, "DGETRI", " ", n, -1, -1, -1))
+		}
+	}
+	ldwork = nb
+	bi := blas64.Implementation()
+	// TODO(btracey): Replace this with a more row-major oriented algorithm.
+	if nb < nbmin || nb >= n {
+		// Unblocked code.
+		for j := n - 1; j >= 0; j-- {
+			for i := j + 1; i < n; i++ {
+				work[i*ldwork] = a[i*lda+j]
```

Comment:
I haven't transposed work. Before, the indexing is effectively work(i,1), and I have transformed that indexing into row major. this particular loop is easy to re-orient, but it's more complicated below in the Dgemv and Dgemm calls.